src/main/scala/rules/SymbolicExecutionRules.scala

```diff
@@ -28,14 +28,18 @@ trait SymbolicExecutionRules {
       val model = v.decider.getModel()
       if (model != null && !model.contains("model is not available")){
         val nativeModel = Model(model)
-        val ce: Counterexample = if (Verifier.config.counterexample.toOption.get == "native") {
-          val oldHeap = if (s.oldHeaps.contains(Verifier.PRE_STATE_LABEL))
-            Some(s.oldHeaps(Verifier.PRE_STATE_LABEL).values)
-          else None
-          SiliconNativeCounterexample(s.g, s.h.values, oldHeap, nativeModel)
-        }else{
-          SiliconVariableCounterexample(s.g, nativeModel)
+        val ce_type = Verifier.config.counterexample()
+        val ce: Counterexample = ce_type match {
+          case "native" => 

```

Comment:
Using an enum or a case object, instead of string literals, would facilitate maintenance

Reply:
Here I am a bit confused. This is the user input, i.e. the provided flag that comes with --counterexample <string> right? So at some point we would have to do this matching. Is your idea to do the string matching when constructing the config such that the config's field "ce_type" is an enum? Or am I completely misunderstanding your point?

Reply:
Sorry, I should have been more explicit, but your guess is right: Scallop has a mechanism for converting "raw" options, e.g. arbitrary strings, into a more constrained form, e.g. values of an enum. This prevents unexpected option values from "leaking" beyond the option parser.

Reply:
Okay but in this case this would have to happen on the silver level right since the counterexample type is parsed in SilFrontEndConfig.scala? Do you want me to open a new PR for that? I got it working but I am still not sure if this is what you expected, I assume this will require changes in Carbon too.

Reply:
Good point. In that case, leave the code as-is, but please add a TODO comment suggesting the change to an enum.